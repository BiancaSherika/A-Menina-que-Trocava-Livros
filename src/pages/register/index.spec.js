import { Register } from './index.js';
import * as services from '../../services/index.js';

services.registerUser = jest.fn(() => Promise.resolve(true));

describe('Register', () => {
  it('should be a function', () => {
    expect(typeof Register).toBe('function');
  });

  it('should load the page', () => {
    expect(Register()).toMatchSnapshot();
  });
  it('when the user clicks the button he must have completed the registration', () => {
    Register().querySelector('#button-register').dispatchEvent(new Event('click'));
    expect(services.registerUser).toHaveBeenCalled();
  });
});
