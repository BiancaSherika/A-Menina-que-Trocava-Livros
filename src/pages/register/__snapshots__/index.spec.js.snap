// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register should load the page 1`] = `
<div>
  
    
  <form
    class="container-register"
  >
    
      
    <h1
      class="paragrafe"
    >
      Crie sua conta
    </h1>
    
      
    <input
      class="input-register"
      id="name"
      placeholder="Nome"
      type="name"
    />
    
      
    <input
      class="input-register"
      id="email"
      placeholder="E-mail"
      type="email"
    />
    
      
    <input
      class="input-register"
      id="password"
      placeholder="Senha"
      type="password"
    />
    
      
    <p
      class="instruction"
    >
      A senha deve ter no mínimo 6 dígitos
    </p>
    
      
    <button
      class="button-register"
      id="button-register"
      type="submit"
    >
      Registrar-se
    </button>
    
    
  </form>
  
  
</div>
`;
